### ISSUE TITLE
Allow server to change player names.

### ISSUE THREAD

**Issue by TsFreddie** [teeworlds/teeworlds#2744]
We should add back the ability for servers to change player names / clans.

Reason being:
- Several mods still need it.
- More importantly, we should add a rcon command to allow admin / mod to change a player's name if it is inappropriate.

**Comment by ChillerDragon** [comment]
Yes please current workaround is reconnecting players and fake ids and ugly hacks like that. Would be nice to avoid that.

**Comment by ChillerDragon** [comment]
Also what about going back to (1) in front of duplicated names. Lets be honest here we tested the id thing but literally nobody looks at the ids. Most players do not even know what these numbers mean. It is technical info that is displayed by default and polluting the HUD. Nobody turns of defaults or pays attention to it. The whole 0.7 new name system was a fail. Humans are bad with numbers. I mentioned the ID issue in a stream once and still was able to successfully fake a player minutes after talking about the id system. Also since there is /whisper names should not be duplicated or otherwise it is hard to pick the correct target.

**Comment by Dune-jr** [comment]
> * More importantly, we should add a rcon command to allow admin / mod to change a player's name if it is inappropriate.

That wouldn't work with older clients

**Comment by TsFreddie** [comment]
> That wouldn't work with older clients

I'm thinking it should be better than not having it at all. It should be enough for new clients to see the changes since I don't see it affecting actual gameplay. But yeah, older clients can't see the name change.

**Comment by TsFreddie** [comment]
Forgot about whispering and highlighting. We should discuss a bit more about this.

**Comment by Dune-jr** [comment]
> > That wouldn't work with older clients
> 
> I'm thinking it should be better than not having it at all. It should be enough for new clients to see the changes since I don't see it affecting actual gameplay. But yeah, older clients can't see the name change.

Eh, moderation commands should work for everyone or people would mod their client to not implement this feature, kinda sucks

**Comment by TsFreddie** [comment]
> Eh, moderation commands should work for everyone or people would mod their client to not implement this feature, kinda sucks

Don't know about older clients, but if people mod their client to be not affected by server name change, the server and everyone else should still sees the name change which is the goal of moderating isn't it? Like you can't dodge the change by just not changing it on your client (and older clients unforturnately).

**Comment by Dune-jr** [comment]
> the server and everyone else should still sees the name change which is the goal of moderating isn't it?

that sounds bad, people would call them by the new name, causing misunderstandings, pings wouldn't work etc.

**Comment by heinrich5991** [comment]
That's only a concern with old clients, not modded clients.

I can modify the client today to show a different name for some people, that wouldn't be a problem of the server.

**Comment by Dune-jr** [comment]
Yes, but modded clients would benefit from faking old clients to shield themselves from this mod action

**Comment by TsFreddie** [comment]
> that sounds bad

is does, for old clients :(

It is still possible for older clients to get other people's name changes using the silent disconnect way, but that sounds like a hacky implementation.

but cases of such misunderstandings should be sparse, i'd imagine the command is not used all the time. Meanwhile mods using it can benefit from this a lot. Just need to put some thought into whether the trade-off is worth it.

> modded clients would benefit from faking old clients

Modded clients can do anything they want already. Their name should still be changed for everyone else (who's using newer clients).

**Comment by heinrich5991** [comment]
> Yes, but modded clients would benefit from faking old clients to shield themselves from this mod action

Not if you treat old clients the same, (creating the misunderstandings you talked about), and just modify their names.

**Comment by Dune-jr** [comment]
> > Yes, but modded clients would benefit from faking old clients to shield themselves from this mod action
> 
> Not if you treat old clients the same, (creating the misunderstandings you talked about), and just modify their names.

Then people would misping old clients, right?

**Comment by TsFreddie** [comment]
#2741 also "desync" names for older clients, and it may even create invalid unicode characters on older clients due to how the names are truncated. ;(

**Comment by Learath2** [comment]
The hacky silent disconnect can be a fallback for old clients. Modded clients to undo mod actions sounds silly. It only shows them the old name, what is the benefit to that?
